### PR TITLE
[AKS] prometheus config leverage service account access api-server

### PR DESCRIPTION
**Pull request by YanjieGao**
prometheus deployment leverage service account. this is maybe useful for deploy pai on existing k8s which api-server access need auth.

when use this on aks:

- user config use_tls
- config api-server url. 
note: this has the dependency with pr (https://github.com/Microsoft/pai/pull/1428)  for api-server url configuration.

### CODE REVIEW

File: src/prometheus/deploy/prometheus-configmap.yaml.template [hunk 1]
```diff
@@ -36,6 +36,13 @@ data:
     scrape_configs:
     - job_name: 'node_exporter'
       scrape_interval: {{ prom_info['scrape_interval']|default(30) }}s
+      {% if clusterinfo['prometheusinfo']['use_tls'] -%}
+      tls_config:
+        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
```
Comment (xudifsd) [inline src/prometheus/deploy/prometheus-configmap.yaml.template]: This path is a volume or what?

Comment (YanjieGao) [reply src/prometheus/deploy/prometheus-configmap.yaml.template]: a volume mounted by k8s for service account auth

Comment (xudifsd) [reply src/prometheus/deploy/prometheus-configmap.yaml.template]: Ok, then, I think it's inappropriate to config under `prometheus`.

Comment (YanjieGao) [reply src/prometheus/deploy/prometheus-configmap.yaml.template]: yes, watchdog's method is a little different with prometheus. it need reconfig http url access.

Comment (YanjieGao) [reply src/prometheus/deploy/prometheus-configmap.yaml.template]: make config as a global cluster wide config which prometheus watchdog webportal future use

**Pull request closed without merging**